kin-sdk/kin-base/src/main/java/kin/base/Transaction.java

```diff
@@ -228,7 +228,14 @@ public static Transaction fromEnvelopeXdr(String envelope) throws IOException {
    */
   public static Transaction fromEnvelopeXdr(TransactionEnvelope envelope) {
     kin.base.xdr.Transaction tx = envelope.getTx();
+    // Although currently there couldn't be a transaction with no operations we still check it.
     int mFee = tx.getFee().getUint32();
+    if (tx.getOperations().length > 1) {
+      // Because the fee was already multiplied by number of operation then divide by it because when reCreate this
+      // transaction then we will multiple it again.
+      mFee = mFee / tx.getOperations().length;
```

Comment:
To make things clearer and less bug prone, I suggest you rename the fee parameter in the `Transaction` constructor to be called `feePerOperation` 
And then you can also change the code here as follows: 
```
int totalFee = tx.getFee().getUint32();
int feePerOperation = 0;
  if (tx.getOperations().length > 1 ){
       feePerOperation = totalFee / tx.getOperations().length
  }
```

Reply:
I may do so, although as i wrote before it is a code that we got from stellar so didn't want to mess with yet too much.